2001 add application and process json files in zip

constantin-ciobotaru opened the pull request.

Commit:
2001 add application and process json files in zip

- add <application-name>.json and <process-name>.json files when export zip
- use the ObjectMapper to create the entities from json
- return 400 when cannot import from file
- don't use anymore the file name as model name
- delete child models before deleting an application
- update the tests for import/export applications
- add tests for import/export models

https://github.com/Activiti/Activiti/issues/2001

erdemedeiros commented on activiti-cloud-services-org/activiti-cloud-services-org-api/src/main/java/org/activiti/cloud/organization/api/ModelType.java:
```diff
@@ -24,4 +24,6 @@
 
     String getFolderName();
+
+    String getContentFileExtension();
```
I'm wondering processes are not going to be a problem as there are two possible extensions `bpmn20.xml` and `**.bpmn`.

erdemedeiros commented on activiti-cloud-services-org/activiti-cloud-services-org-api/src/main/java/org/activiti/cloud/organization/api/ProcessModelType.java:
```diff
@@ -28,6 +30,8 @@
 
     public static final String PROCESSES = "processes";
 
+    public static final String BPMN20_XML = "bpmn20.xml";
```
As my previous comment `**.bpmn` is also a valid extension for processes (maybe legacy only, I'm not sure).

constantin-ciobotaru replied:
I've added allowedFileExtensions as string array of "bpmn20.xml", "bpmn.xml" and "bpmn". I hope, this covers all possibilities.

erdemedeiros commented on activiti-cloud-services-org/activiti-cloud-services-org-rest/src/test/java/org/activiti/cloud/services/organization/rest/controller/ApplicationControllerIT.java:
```diff
@@ -325,13 +331,33 @@ public void testExportApplication() throws Exception {
+                        "processes/process-model-1.json",
+                        "processes/process-model-2.bpmn20.xml",
+                        "processes/process-model-2.json")
+                .hasJsonContentSatisfying("application-with-models.json",
```
@constantin-ciobotaru just to let you know, for json assertions we have been using [JsonUnit](https://github.com/lukas-krecan/JsonUnit#fluent) with which you can write fluent assertions in similar way to AssertJ.

constantin-ciobotaru replied:
Moved to json-unit-fluent. Thanks.

erdemedeiros reviewed (commented):
@constantin-ciobotaru have some question before merging. Please, take a look on my comments.
Also, we definitely need to find a better solution than committing zip files.

Commit:
2001 add allowed file extensions for model

- add allowed file extensions in ModelType
- use json-unit-fluent instead json-path-assert in json assertions

https://github.com/Activiti/Activiti/issues/2001
